l1-contracts/src/core/messagebridge/Inbox.sol

```diff
@@ -89,7 +89,11 @@ contract Inbox is IInbox {
     });
 
     bytes32 leaf = message.sha256ToField();
-    uint256 index = currentTree.insertLeaf(leaf);
+    // this is the global leaf index and not index in the l2Block subtree
+    // such that users can simply use it and don't need access to a node if they are to consume it in public.
+    // trees are constant size so global index = tree number * size + subtree index
+    uint256 index =
+      (inProgress - Constants.INITIAL_L2_BLOCK_NUM) * SIZE + currentTree.insertLeaf(leaf);
     totalMessagesInserted++;
     emit MessageSent(inProgress, index, leaf);
```

Comment:
having the L2Block number here is redundant information now as it can be derived from the index. For this reason I would nuke it from the event.

Reply:
Spoke with Lasse and we think we should keep the l2block in the event as it helps people search for their index on L1!

Reply:
also helps with alpha team to make messages unique